roofit/roofitcore/src/TestStatistics/LikelihoodWrapper.cxx

```diff
@@ -0,0 +1,159 @@
+// Author: Patrick Bos, Netherlands eScience Center / NIKHEF 2021
+
+/*****************************************************************************
+ * RooFit
+ * Authors:                                                                  *
+ *   WV, Wouter Verkerke, UC Santa Barbara, verkerke@slac.stanford.edu       *
+ *   DK, David Kirkby,    UC Irvine,         dkirkby@uci.edu                 *
+ *                                                                           *
+ * Copyright (c) 2000-2021, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+
+#include <TestStatistics/LikelihoodWrapper.h>
+#include <TestStatistics/RooAbsL.h> // need complete type for likelihood->...
+#include <TestStatistics/MinuitFcnGrad.h>
+#include <TestStatistics/RooUnbinnedL.h>
+#include <TestStatistics/RooSumL.h> // need complete type for dynamic cast
+
+namespace RooFit {
+namespace TestStatistics {
+
+LikelihoodWrapper::LikelihoodWrapper(std::shared_ptr<RooAbsL> likelihood,
+                                     std::shared_ptr<WrapperCalculationCleanFlags> calculation_is_clean/*,
+                                     RooMinimizer *minimizer*/)
+   : likelihood_(std::move(likelihood)),/* minimizer_(minimizer),*/
+     calculation_is_clean_(std::move(calculation_is_clean)) /*, minimizer_fcn_(minimizer_fcn)*/
+{
+   // Note to future maintainers: take care when storing the minimizer_fcn pointer. The
+   // RooAbsMinimizerFcn subclasses may get cloned inside MINUIT, which means the pointer
+   // should also somehow be updated in this class.
+}
+
+void LikelihoodWrapper::synchronizeWithMinimizer(const ROOT::Math::MinimizerOptions & /*options*/) {}
+
+void LikelihoodWrapper::synchronizeParameterSettings(
+   const std::vector<ROOT::Fit::ParameterSettings> & /*parameter_settings*/)
+{
+}
+
+void LikelihoodWrapper::constOptimizeTestStatistic(RooAbsArg::ConstOpCode opcode, bool doAlsoTrackingOpt)
+{
+   likelihood_->constOptimizeTestStatistic(opcode, doAlsoTrackingOpt);
+}
+
+double LikelihoodWrapper::defaultErrorLevel() const
+{
+   return likelihood_->defaultErrorLevel();
+}
+std::string LikelihoodWrapper::GetName() const
+{
+   return likelihood_->GetName();
+}
+std::string LikelihoodWrapper::GetTitle() const
+{
+   return likelihood_->GetTitle();
+}
```

Comment:
Doxygen and maybe can be inlined (unless it's only forward decls in the header).